include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -528,6 +512,26 @@ SECTIONS
 	KEEP(*(.gnu.attributes))
 	}
 
+#if defined CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS
```

Comment:
Moving the SG section at the end of program area should be fine. 

Now, why should we hard-code the NSC area start and not allow the user to select it themselves? I know that the nRF security attribution unit is somewhat peculiar on how it may configure NSC flash regions - still I would like to see how to address these peculiarities, while, at the same, time allowing the user to have some control over it.

Reply:
Fair point, how about something like this?

	#if CONFIG_ARM_NSC_REGION_BASE_ADDRESS != 0
		#define NSC_ALIGN . = CONFIG_ARM_NSC_REGION_BASE_ADDRESS 
	#elif defined(CONFIG_SOC_SERIES_NRF91X)
		#define NSC_ALIGN . = ALIGN(0x8000) - (1 << LOG2CEIL(__sg_size))
	#else
		#define NSC_ALIGN . = ALIGN(4)
	#endif

	#ifdef CONFIG_SOC_SERIES_NRF91X
		#define NSC_ALIGN_END . = ALIGN(0x8000)
	#else
		#define NSC_ALIGN_END . = ALIGN(4)
	#endif

Reply:
Changed to above suggestion